109-custom-client-authentication.md

```diff
@@ -0,0 +1,156 @@
+It will have a flag indicating whether the authentication mechanism uses SASL.
+And it will have a `config` section for adding additional configuration options such as `sasl.mechanism` or JAAS configuration.
+
+The `secrets` field that is present in the server-side `custom` authentication but is already deprecated will not be used in the client-side API.
+Users can use the _additional volumes_ feature to add custom secrets required by the authentication mechanism.
+
+Users can use configuration providers for values such as passwords to load them from files or environment variables without having them hardcoded and visible in the Strimzi custom resources.
+
+The `config` field will allow configuration of the fields starting with the following prefixes:
+* `ssl.keystore.`
+* `sasl.`
+
+These options are otherwise forbidden and cannot be set in the regular _configuration_ sections (i.e. in `.spec.config` or `.spec.clusters.config`).
+Other options which might be needed by the custom authentication mechanisms can be configured in the regular _configuration_ sections.
+That way, we make sure that users are able to configure the custom mechanisms, but cannot use it to override the other configuration options blocked by Strimzi (e.g. the REST API configuration for Connect) by mistake or intention.
+
+Use of TLS and the configuration of trusted certificates will continue to be done through the corresponding `tls` sections.
+It will not be configurable through the custom authentication mechanism.
+
+### Examples
+
+The following examples show how the new `custom` type authentication might be used for various use cases.
+
+* Amazon IAM authentication for connecting to Amazon MSK:
+  ```yaml
+  authentication:
+    type: custom
+    sasl: true
+    config:
+      sasl.mechanism: AWS_MSK_IAM
+      sasl.jaas.config: software.amazon.msk.auth.iam.IAMLoginModule required;
+      sasl.client.callback.handler.class: software.amazon.msk.auth.iam.IAMClientCallbackHandler
+  ```
+* Microsoft Event Hubs authentication using Microsoft Entra ID:
+  ```yaml
+  authentication:
+    type: custom
+    sasl: true
+    config:
+      sasl.mechanism: OAUTHBEARER
+      sasl.jaas.config: org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required;
+      sasl.login.callback.handler.class: CustomAuthenticateCallbackHandler
+  ```
+* Custom mTLS authentication (useful for example when the key is injected by some custom mechanism instead of being loaded from a Kubernetes Secret):
+  ```yaml
+  authentication:
+    type: custom
+    sasl: false
+    config:
+      ssl.keystore.location: /mnt/my-custom-tls-cert/keystore.p12
+      ssl.keystore.password: changeme
+      ssl.keystore.type: PKCS12
+  ```
+
+### Custom authentication plugins
+
+Strimzi will not include any custom authentication plugins in its container images.
+Users will be responsible for adding them on their own.
```

Comment:
we don't need to say anything on where they'll reside or how they're found?

Reply:
I added a link to the docs which cover this: https://strimzi.io/docs/operators/latest/full/configuring.html#con-common-configuration-volumes-reference